custom/inference.py

```diff
@@ -46,30 +46,63 @@ def print_fps(self):
             print(f"Inference running at {self.count / self.tottime:.3f} FPS")
             self.last_print = time.time()
 
+
+
+
 class Inference():
     def __init__(self,
-                 config_file=config_file,
-                 checkpoint_file=checkpoint_file,
-                 inf_dir=inf_dir,
-                 inf_out_dir=inf_out_dir):
+                 config_file,
+                 checkpoint_file,
+                 inf_dir,
+                 inf_out_dir,
+                threshold=50,
+                output_file = None,
+                 ):
         self.model = init_detector(config_file, checkpoint_file, device='cuda:0')
         os.makedirs(inf_out_dir, exist_ok=True)
         self.inf_dir = inf_dir
         self.inf_out_dir = inf_out_dir
+        self.output_file = output_file
         self.fps_logger = FPSLogger()
-        self.run()
+        self.run(threshold)
+
 
 
-    def run(self):
-        for imn in os.listdir(self.inf_dir):
-            img = f'{inf_dir}/{imn}'
-            self.fps_logger.start_record()
-            result = inference_detector(self.model, img)
-            self.fps_logger.end_record()
-            # model.show_result(img, result)
-            self.model.show_result(img, result, out_file=f'{self.inf_out_dir}/{imn}', score_thr=0.5)
+    
+    def run(self, threshold = 50):
+            results = []
+            for imn in os.listdir(self.inf_dir):
+                img = f'{self.inf_dir}/{imn}'
+                self.fps_logger.start_record()
+                result = inference_detector(self.model, img)
+                results.append({img : extract_bounding_boxes(result, threshold)})
+                self.fps_logger.end_record()
+                # model.show_result(img, result)
+                # self.model.show_result(img, result, out_file=f'{self.inf_out_dir}/{imn}', score_thr=0.5)
```

Comment:
we should keep the option to visualize the images. You can add an optional argument `output_image_dir` that, if given, writes the images there.

Reply:
we already have the inf_out_dir, i will use that